File: LEM_Pilas/Pila_Listas.cs
using System;

public class Nodo
{
    public int data;
    public Nodo next;

    public Nodo(int d)
    {
        data = d;
        next = null;
    }
}

public class Pila_Listas
{
    private Nodo top;

    public Pila_Listas()
    {
        top = null;
    }

    public bool IsEmpty()
    {
        return top == null;
    }

    public void Push(int data)
    {
        Nodo nuevo = new Nodo(data);
        nuevo.next = top;
        top = nuevo;
    }

    public int Pop()
    {
        if (IsEmpty())
        {
            Console.WriteLine("Error: Underflow de Pila");
            return -1;
        }
        int dato = top.data;
        top = top.next;
        return dato;
    }

    public int Peek()
    {
        if (IsEmpty())
        {
            Console.WriteLine("La Pila esta vacia");
            return -1;
        }
        return top.data;
    }

    public static void Main(string[] args)
    {
        Pila_Listas pila = new Pila_Listas();
        pila.Push(10);
        pila.Push(20);
        pila.Push(30);

        Console.WriteLine("Elemento superior: " + pila.Peek());
        Console.WriteLine("Extrae elemento: " + pila.Pop());
        Console.WriteLine("Nuevo elemento superior: " + pila.Peek());
    }
}

File: LEM_Pilas/Pila_Listas.js
class Nodo {
    constructor(d) {
        this.data = d;
        this.next = null;
    }
}

class Pila_Listas {
    constructor() {
        this.top = null;
    }

    isEmpty() {
        return this.top === null;
    }

    push(data) {
        const nuevo = new Nodo(data);
        nuevo.next = this.top;
        this.top = nuevo;
    }

    pop() {
        if (this.isEmpty()) {
            console.log("Error: Underflow de Pila");
            return -1;
        }
        const dato = this.top.data;
        this.top = this.top.next;
        return dato;
    }

    peek() {
        if (this.isEmpty()) {
            console.log("La Pila esta vacia");
            return -1;
        }
        return this.top.data;
    }
}

// --- Ejecución (Simulación de main) ---

const pila = new Pila_Listas();
pila.push(10);
pila.push(20);
pila.push(30);

console.log("Elemento superior: " + pila.peek());
console.log("Extrae elemento: " + pila.pop());
console.log("Nuevo elemento superior: " + pila.peek());